server/conn.go

```diff
@@ -2449,16 +2438,24 @@ func (cc *clientConn) writeChunksWithFetchSize(ctx context.Context, rs ResultSet
 		if err = cc.writePacket(data); err != nil {
 			return err
 		}
+
+		iter.Next()
 	}
-	if stmtDetail != nil {
-		stmtDetail.WriteSQLRespDuration += time.Since(start)
+	if iter.Error() != nil {
+		return iter.Error()
+	}
+
+	// tell the client COM_STMT_FETCH has finished by setting proper serverStatus,
+	// and close ResultSet.
+	if iter.Current() == iter.End() {
+		serverStatus &^= mysql.ServerStatusCursorExists
+		serverStatus |= mysql.ServerStatusLastRowSend
 	}
+
 	if cl, ok := rs.(fetchNotifier); ok {
 		cl.OnFetchReturned()
```

Comment:
So `OnFetchReturned` is counted in `WriteSQLRespDuration`, too?

Reply:
Nice catch. It should be excluded. Fixed now.